core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -2416,6 +2416,24 @@ LogicalProject(X=[+($0, $1)], B=[$1], A=[$0])
         <Resource name="planAfter">
             <![CDATA[
 LogicalValues(tuples=[[{ 11, 1, 10 }, { 23, 3, 20 }]])
+]]>
+        </Resource>
+    </TestCase>
+    <TestCase name="testReduceValuesNull">
+        <Resource name="sql">
+            <![CDATA[insert into sales.dept(deptno,name) values (0, 'null')]]>
+        </Resource>
+        <Resource name="planBefore">
+            <![CDATA[
+LogicalTableModify(table=[[CATALOG, SALES, DEPT]], operation=[INSERT], flattened=[false])
```

Comment:
I can see what's being tested here, but not sure if it is the original intention of that ignored test as you have added a SELECT there. I guess the original test is used to test if NULL works when applying `ValuesReduceRule`, and the new test does not include it.

Will need a committer who is more familiar with optimization to take a look then.

Reply:
I update the PR, add the NULL case.